scripts/install-du.ps1

```diff
@@ -94,30 +97,71 @@ function CalcHash {
     [Convert]::ToBase64String($hashAlg.ComputeHash([System.IO.File]::ReadAllBytes($FilePath)))
 }
 
-function WriteExtensionRegistration {
+function WriteContentDownloaderRegistration{
     Param(
-        [Parameter(Mandatory = $true)][string]$BinaryFilePath,
-        [Parameter(Mandatory = $true)][string]$OutputJsonFilePath
+        [Parameter(Mandatory = $true)][string]$BinaryFileNameNoExt,
+        [Parameter(Mandatory = $true)][string]$ExtensionsDir
     )
 
-    $fileSizeInBytes = (Get-Item $BinaryFilePath).Length
-    $fileHash = CalcHash -FilePath $BinaryFilePath
+    $adu_extensions_sources_dir = "$ExtensionsDir/sources"
+    $binaryFilePath = "$adu_extensions_sources_dir/${BinaryFileNameNoExt}.dll"
+
+    $fileSizeInBytes = (Get-Item $binaryFilePath).Length
+    $fileHash = CalcHash -FilePath $binaryFilePath
 
     $registration = [ordered]@{
-        fileName="$BinaryFilePath"
-        sizeInBytes=$FileSizeInBytes
+        fileName="$binaryFilePath"
+        sizeInBytes=$fileSizeInBytes
         hashes = @{
             sha256="$fileHash"
         }
     }
 
-    $registration | ConvertTo-Json | Set-Content $OutputJsonFilePath
+    $registration | ConvertTo-Json | Set-Content "$ExtensionsDir/content_downloader/extensions.json"
+}
+
+function WriteContentHandlerRegistration{
+    Param(
+        [Parameter(Mandatory = $true)][string]$BinaryFileNameNoExt,
+        [Parameter(Mandatory = $true)][string]$ExtensionName,
+        [Parameter(Mandatory = $true)][string]$ExtensionsDir
+    )
+
+    $adu_extensions_sources_dir = "$ExtensionsDir/sources"
+
+    $binaryFilePath = "$adu_extensions_sources_dir/${BinaryFileNameNoExt}.dll"
+    $fileSizeInBytes = (Get-Item $binaryFilePath).Length
+    $fileHash = CalcHash -FilePath $binaryFilePath
+
+    $numUnderscore = ($ExtensionName | Select-String -Pattern '_' -AllMatches).Matches.Count
```

Comment:
not clear to me why this is needed?

Reply:
in any case, as I mention above, we shouldn't be creating our own registration code -- let's change the registration from self-registration to (or in addition?) outputting json

Reply:
It's because the handlerId for "update manifest" handler only has one underscore, that is, no `"_1"` version at the end: e.g. `"microsoft_update-manifest"` which will get converted to `"microsoft/update-manifest"`

The handlerId of "step handlers" always have 2 underscores because they are always versioned:
`"microsoft_apt_1" -> "microsoft/apt:1`